packages/shared/src/lib/func.ts

```diff
@@ -144,3 +151,13 @@ export const initReactModal = ({
 
   globalThis.reactModalInit = true;
 };
+
+export const initApp = (): void => {
+  const param = globalThis?.location?.search;
+  if (param?.includes('android=true')) {
+    globalThis.isAndroidApp = true;
+  }
+  if (param?.includes('pwa=true')) {
+    globalThis.isPWA = true;
+  }
```

Comment:
Yeah this might get cleared on full page reload, for example after oauth flow.

Reply:
@capJavert An alternative would be to add it to localStorage instead I guess. What do you think?

Reply:
Issue is that anyone could add those params and then start to act as if they were in android app or pwa, it would be much better if we could find a way to use feature detection instead. Did you try to look into if we can detect pwa and android app through some specific browser features/APIs they have?

Reply:
Yes, I did. Didn't seem to be any real common consensus on the best way to do it. For the Android app, we could alternatively set a custom user agent in the Java code before launching the webview, but seems like a lot of effort for not a lot of gain.

Maybe I'm not seeing the bigger picture, but not sure why anyone would be editing those, or what the benefit of doing it would be? The purpose of the value is purely UX

Reply:
I think its always good to look at it from perspective of someone messing with it 😆 

Because in the future we might rely on this to do some more crucial or collect analytics data and then the data will not be accurate due to above method not being reliable. 

For android it seems like a good way to inject custom user agent since we can control it and injecting user agent is much harder then adjusting the URL in browser.

For PWA I did a google and there is this method, did you try it already?
```js
const isIos = /iPhone|iPad|iPod/.test(navigator.userAgent);
const isInStandaloneMode = window.navigator.standalone || window.matchMedia('(display-mode: standalone)').matches;

if (isIos && isInStandaloneMode) {
    console.log('PWA on iOS detected');
}
```

Seems to work if I test it with current PWA.

That said your implementation might be acceptable but I think all stakeholders and product need to be made aware of the downsides.

Reply:
I agree with the iOS PWA. That snippet from Ante works well. Not sure though when it comes to Android.

Reply:
Most I see from the internet also checks the user agent when it comes to Android. Probably a helpful resource: https://stackoverflow.com/questions/77982241/how-can-i-detect-if-im-in-an-app-or-im-in-a-browser-on-android-with-javascript

Reply:
> I think its always good to look at it from perspective of someone messing with it 😆
> 
> Because in the future we might rely on this to do some more crucial or collect analytics data and then the data will not be accurate due to above method not being reliable.
> 
> For android it seems like a good way to inject custom user agent since we can control it and injecting user agent is much harder then adjusting the URL in browser.
> 
> For PWA I did a quick google and there is this method, did you try it already?
> 
> ```js
> const isIos = /iPhone|iPad|iPod/.test(navigator.userAgent);
> const isInStandaloneMode = window.navigator.standalone || window.matchMedia('(display-mode: standalone)').matches;
> 
> if (isIos && isInStandaloneMode) {
>     console.log('PWA on iOS detected');
> }
> ```
> 
> Seems to work if I test it with current PWA.
> 
> That said your implementation might be acceptable but I think all stakeholders and product need to be made aware of the downsides.

I did see this myself, but when I was looking at this yesterday, it seemed a bit concerning that it doesn't even have a proper entry into MDN:
![Screenshot 2024-11-28 at 13 57 24](https://github.com/user-attachments/assets/3715563f-09c0-4c7b-9556-193c00d1c655)

Seems a tad better than the check for ?pwa=true since the browser sets it itself, but going the localStorage route seems more ideal to keep the check for PWA and App more consistent. We can decide on the PWA approach in Slack sinceI  preemptively added this due to the iOS onboarding being the next task. Not crucial for this task.

I think adjusting the user agent from the Java code is a bit out of scope for this experiment as well. Wouldn't you agree?  

@sshanzel We have a function called `isWebView` that covers that user agent. I did bring it up in guild-web, but I'm not entirely sure if it's reliable enough, since webview can be used for a variety of things. Could be a good temporary implementation for now, though.